### PR TITLE
K8SPXC-751 cross-site replication doc

### PR DESCRIPTION
**Pull request by fiowro**
[K8SPXC-751](https://jira.percona.com/browse/K8SPXC-751) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

File: source/replication.rst [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+* ``pxc.replicationChannels.[].isSource`` key should be set to ``false``,
+
+* ``pxc.replicationChannels.[].sourcesList`` is the list of *Source* cluster names from which Replica should get the data,
+
+* ``pxc.replicationChannels.[].sourcesList.[].host`` is the host name or IP-address of the Source,
+
+* ``pxc.replicationChannels.[].sourcesList.[].port`` is the port of the source (``3306`` port will be used if nothing specified),
+
+* ``spec.pxc.replicationChannels.[].sourcesList.[].weight`` is the *weight* of the source (``100`` by default).
```
Comment (qJkee) [inline source/replication.rst:70]: remove spec =)

File: source/replication.rst [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+       expose:
+         enabled: true
+         type: LoadBalancer
+         loadBalancerSourceRanges:
+           - 10.0.0.0/8
+         annotations: 
+           networking.gke.io/load-balancer-type: "Internal"
+
+.. note:: This will create the internal LoadBalancer per each Percona XtraDB
+   Cluster node.
```
Comment (qJkee) [inline source/replication.rst:124]:
```suggestion
   Cluster pod.
```

**Review by spron-in — CHANGES_REQUESTED** [review]
I think we should move "Exposing instances of Percona XtraDB Cluster" under "Configuring cross-site replication on Source instances". It is how the user is going to read and configure - first configure the replication channels on the Source and expose the Source pods as the next step.

**Review by hors — APPROVED** [review]

**Review by qJkee — DISMISSED** [review]

File: source/replication.rst [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+The cross-site replication involves configuring one Percona XtraDB Cluster as *Source*, and another Percona XtraDB Cluster as *Replica* to allow an asynchronous replication between them:
+
+ .. image:: ./assets/images/pxc-replication.svg
+   :align: center
+
+The Operator automates configuration of *Source* and *Replica* Percona XtraDB Clusters, but the feature itself is not bound to Kubernetes. Either *Source* or *Replica* can run outside of Kubernetes and be out of the Operators’ control. 
+
+This feature can be useful in several cases: for example, it can simplify migration from on-premises to the cloud with replication, and it can be really helpful in case of the disaster recovery too.
+
+.. note:: Cross-site replication is based on `Automatic Asynchronous Replication Connection Failover<https://dev.mysql.com/doc/refman/8.0/en/replication-asynchronous-connection-failover.html>`_. Therefore it requires  MySQL 8.0 (Percona XtraDB Cluster 8.0) to work.
```
Comment (spron-in) [inline source/replication.rst:15]:
```suggestion
.. note:: Cross-site replication is based on `Automatic Asynchronous Replication Connection Failover<https://dev.mysql.com/doc/refman/8.0/en/replication-asynchronous-connection-failover.html>`_. Therefore it requires  MySQL 8.0.22+ (Percona XtraDB Cluster 8.0.22+) to work.
```

File: source/replication.rst [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+Set up Percona XtraDB Cluster cross-site replication
+====================================================
+
+The cross-site replication involves configuring one Percona XtraDB Cluster as *Source*, and another Percona XtraDB Cluster as *Replica* to allow an asynchronous replication between them:
+
+ .. image:: ./assets/images/pxc-replication.svg
+   :align: center
+
+The Operator automates configuration of *Source* and *Replica* Percona XtraDB Clusters, but the feature itself is not bound to Kubernetes. Either *Source* or *Replica* can run outside of Kubernetes and be out of the Operators’ control.
```
Comment (spron-in) [inline source/replication.rst:11]:
```suggestion
The Operator automates configuration of *Source* and *Replica* Percona XtraDB Clusters, but the feature itself is not bound to Kubernetes. Either *Source* or *Replica* can run outside of Kubernetes, be regular MySQL and be out of the Operators’ control. 
```

Comment (spron-in) [reply source/replication.rst:11]: I believe it is important to mention that this feature can work for regular MySQL as well, not only PXC.

File: source/replication.rst [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+The Operator automates configuration of *Source* and *Replica* Percona XtraDB Clusters, but the feature itself is not bound to Kubernetes. Either *Source* or *Replica* can run outside of Kubernetes and be out of the Operators’ control. 
+
+This feature can be useful in several cases: for example, it can simplify migration from on-premises to the cloud with replication, and it can be really helpful in case of the disaster recovery too.
+
+.. note:: Cross-site replication is based on `Automatic Asynchronous Replication Connection Failover<https://dev.mysql.com/doc/refman/8.0/en/replication-asynchronous-connection-failover.html>`_. Therefore it requires  MySQL 8.0 (Percona XtraDB Cluster 8.0) to work.
+
+.. Describe how to stop/start replication
+   Describe how to perform a failover
+
+Setting up Percona XtraDB Cluster for asynchronous replication without the Operator is described `here <https://www.percona.com/blog/2018/03/19/percona-xtradb-cluster-mysql-asynchronous-replication-and-log-slave-updates/>`_ and is out of the scope for this document.
```
Comment (spron-in) [inline source/replication.rst:20]: This is quite an old blog post. Shall we use this one: https://www.percona.com/blog/2021/04/14/what-you-can-do-with-auto-failover-and-percona-distribution-for-mysql-8-0-x/ instead?

**Review by spron-in — CHANGES_REQUESTED** [review]

**Review by qJkee — APPROVED** [review]

File: source/replication.rst [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+   spec:
+     pxc:
+       expose:
+         enabled: true
+         type: LoadBalancer
+         loadBalancerSourceRanges:
+           - 10.0.0.0/8
+         annotations: 
+           networking.gke.io/load-balancer-type: "Internal"
```
Comment (spron-in) [inline source/replication.rst:72]: we say below " Please take into account, that you will need this IP address to be external
   for cross-region replication in most cases."

why do we have internal load balancer in the example? :)

**Review by qJkee — APPROVED** [review]

**Review by spron-in — APPROVED** [review]

**Pull request merged**